include/time_series/time_series.hpp

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file time_series.hpp
+ * @author Vincent Berenz
+ * license License BSD-3-Clause
+ * @copyright Copyright (c) 2019, Max Planck Gesellshaft.
+ */
+
+#pragma once
+
+#include <chrono>
+#include <cmath>
+
+#include "real_time_tools/timer.hpp"
+
+// virtual class specifying all functions
+// a time_series class should implement
+// Defines also Index and Timestamp
+#include "time_series/interface.hpp"
+
+// all common code to TimeSeries and
+// multiprocesses TimeSeries
+#include "time_series/internal/base.hpp"
+
+// Mutex, ConditionVariable, Lock and Vector
+// use different implementation for TimeSeries and
+// MultiprocessesTimeSeries. Those are defined there.
+#include "time_series/internal/specialized_classes.hpp"
+
+namespace time_series
+{
+/**
+ * @brief Threadsafe time series
+ */
+template <typename T = int>
+class TimeSeries : public internal::TimeSeriesBase<internal::SINGLEPROCESS, T>
```

Comment:
Typically thread safe API are not explicitly called "MultiThreadXXX", because thread safe is only one of the aspect of the API. I feel it clutters the name a bit.  I would vote against, but I do not have a strong opinion.